### PR TITLE
ci: improve action build process

### PR DESCRIPTION
**Pull request by lwjohnst86**
Right now there are three Actions that run all at once. They often fail, which means I get three emails saying they failed. It makes more sense that they run in sequence, so I converted the existing ones into reusable actions (`workflow_call:`) and created one that sets them all in sequence (first `check`, then `coverage`, then `deploy`).

### CODE REVIEW

File: .github/workflows/build-package.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+jobs:
+  check:
+    uses: .github/workflows/R-CMD-check.yaml@main
+
+  coverage:
+    uses: .github/workflows/test-coverage.yaml@main
+    needs: check
+
+  deploy:
+    uses: .github/workflows/pkgdown.yaml@main
```
Comment (lwjohnst86) [inline .github/workflows/build-package.yaml:23]:
```suggestion
    uses: ./.github/workflows/R-CMD-check.yaml@main

  coverage:
    uses: ./.github/workflows/test-coverage.yaml@main
    needs: check

  deploy:
    uses: ./.github/workflows/pkgdown.yaml@main
```

File: .github/workflows/build-package.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+jobs:
+  check:
+    uses: ./.github/workflows/R-CMD-check.yaml@main
+
+  coverage:
+    uses: ./.github/workflows/test-coverage.yaml@main
+    needs: check
+
+  deploy:
+    uses: ./.github/workflows/pkgdown.yaml@main
```
Comment (lwjohnst86) [inline .github/workflows/build-package.yaml:23]:
```suggestion
    uses: ./.github/workflows/R-CMD-check.yam

  coverage:
    uses: ./.github/workflows/test-coverage.yaml
    needs: check

  deploy:
    uses: ./.github/workflows/pkgdown.yaml
```

File: .github/workflows/build-package.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+jobs:
+  check:
+    uses: ./.github/workflows/R-CMD-check.yaml
+
+  coverage:
+    uses: ./.github/workflows/test-coverage.yaml
+    needs: check
+
+  deploy:
```
Comment (lwjohnst86) [inline .github/workflows/build-package.yaml:22]:
```suggestion
  deploy:
    permissions:
      contents: write
```

**Review by Aastedet — APPROVED** [review]

**Review by signekb — APPROVED** [review]
Makes sense 👍

**Pull request merged**